shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/NetworkHealthCheckImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.redhat.service.bridge.shard.operator;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Default;
+
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.Readiness;
+
+@Readiness
```

Comment:
@r00ta Does a failing readiness check on the Operator have any impact? My understanding is that readiness failures would remove this `Pod` from those matched by the selector on a `Service`, so that traffic is not sent to a Pod that is not "ready" to handle traffic.  We don't have any `Service` definition for the Operator (it doesn't need it), so does the readiness failure actually do anything?

Is it better for us to expose metrics instead around number of failed updates to the Manager and then alert when they exceed a particular threshold over time?

Reply:
We have a readiness check in the k8s Deployment https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/0c01459020071793c3151c1834ee408e09b5e5d1/kustomize/base/shard/resources/deploy.yaml#L30 . So if the application is marked as not ready, we get notified and the deployment will be marked as unhealthy